packages/contracts/test/exchange/core.ts

```diff
@@ -481,6 +482,118 @@ describe('Exchange core', () => {
+            expect(orderInfo.orderTakerAssetFilledAmount).to.be.bignumber.equal(expectedTakerAssetFilledAmount);
+            expect(orderInfo.orderStatus).to.equal(expectedOrderStatus);
+        });
+        it('should return the correct orderInfo for a partially filled order', async () => {
+            const takerAssetFillAmount = signedOrder.takerAssetAmount.div(2);
+            await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress, { takerAssetFillAmount });
+            const orderInfo = await exchangeWrapper.getOrderInfoAsync(signedOrder);
+            const expectedOrderHash = orderHashUtils.getOrderHashHex(signedOrder);
+            const expectedTakerAssetFilledAmount = takerAssetFillAmount;
+            const expectedOrderStatus = OrderStatus.FILLABLE;
+            expect(orderInfo.orderHash).to.be.equal(expectedOrderHash);
+            expect(orderInfo.orderTakerAssetFilledAmount).to.be.bignumber.equal(expectedTakerAssetFilledAmount);
+            expect(orderInfo.orderStatus).to.equal(expectedOrderStatus);
+        });
+        it('should return the correct orderInfo for a cancelled and unfilled order', async () => {
+            await exchangeWrapper.cancelOrderAsync(signedOrder, makerAddress);
+            const orderInfo = await exchangeWrapper.getOrderInfoAsync(signedOrder);
+            const expectedOrderHash = orderHashUtils.getOrderHashHex(signedOrder);
+            const expectedTakerAssetFilledAmount = new BigNumber(0);
+            const expectedOrderStatus = OrderStatus.CANCELLED;
+            expect(orderInfo.orderHash).to.be.equal(expectedOrderHash);
+            expect(orderInfo.orderTakerAssetFilledAmount).to.be.bignumber.equal(expectedTakerAssetFilledAmount);
+            expect(orderInfo.orderStatus).to.equal(expectedOrderStatus);
+        });
+        it('should return the correct orderInfo for a cancelled and partially filled order', async () => {
+            const takerAssetFillAmount = signedOrder.takerAssetAmount.div(2);
+            await exchangeWrapper.fillOrderAsync(signedOrder, takerAddress, { takerAssetFillAmount });
+            await exchangeWrapper.cancelOrderAsync(signedOrder, makerAddress);
+            const orderInfo = await exchangeWrapper.getOrderInfoAsync(signedOrder);
+            const expectedOrderHash = orderHashUtils.getOrderHashHex(signedOrder);
+            const expectedTakerAssetFilledAmount = takerAssetFillAmount;
+            const expectedOrderStatus = OrderStatus.CANCELLED;
+            expect(orderInfo.orderHash).to.be.equal(expectedOrderHash);
+            expect(orderInfo.orderTakerAssetFilledAmount).to.be.bignumber.equal(expectedTakerAssetFilledAmount);
+            expect(orderInfo.orderStatus).to.equal(expectedOrderStatus);
+        });
+        it('should return the correct orderInfo for an expired and unfilled order', async () => {
+            await increaseTimeAndMineBlockAsync(signedOrder.expirationTimeSeconds.toNumber());
```

Comment:
`expirationTimeSeconds` is a timestamp, denoted as milliseconds since the unix epoch, whereas `increaseTimeAndMineBlockAsync` expects a duration in seconds. I think what you meant to do was this:

```typescript
const timeOffset = signedOrder.expirationTimeSeconds
    .sub(Date.now())
    .divToInt(1000)
    .toNumber();
await increaseTimeAndMineBlockAsync(timeOffset);
```

Reply:
It was somewhat lazy, but increasing the time by the actual expiration timestamp does guarantee that the order will be expired and also shouldn't affect anything else.

Reply:
There is no way to reverse a time offset and there is a limit to the total time offset that can be applied, so it actually can matter if the offset is way too big. (We're talking about ~1.5 trillion seconds when it should be ~600 seconds, so it really is a huge difference).

In addition, if we're trying to test that order expiration is working correctly, we should be testing that as accurately as possible. The expiration time goes through a lot of conversions and is not hard to imagine that there could be a bug somewhere (e.g. we forgot to convert seconds to milliseconds). As written, the test would pass even if there was a bug causing the expiration time to be off by 10 orders of magnitude.

I'm okay with adding a small buffer here, but not one that is this big.